cmd/crane/rebase.md

```diff
@@ -0,0 +1,121 @@
+This command rewrites an image's manifest to replace layers in a base image
+with layers in another version of that base image. It does so entirely with API
+calls to the registry, so it doesn't have to download or upload full layer
+blobs, if it can be avoided.
+
+This can be useful if you want to produce container images that pick up
+security or bug fixes in base images, without having to completely rebuild the
+image from source. For instance, you might not have access to the original
+source anymore, or you want to produce updated images without performing a full
+rebuild.
+
+*WARNING:* The image that results from such a rebase might not be valid in all
+cases. More details below, but caveat emptor.
+
+## Using `crane rebase`
+
+For purposes of illustration, imagine you've built a container image
+`gcr.io/my-project/my-app:latest`, containing your app, and based on some OS
+image, for instance, `ubuntu`. A vulnerability has been found in the base
+image, and a new fixed version has been released.
+
+You could build your app image again, and its `FROM ubuntu` directive would
+pick up the new base image release, but that requires a full rebuild of your
+entire app from source, which might pull in other changes in dependencies. You
+just want to release this critical bug fix, as quickly as possible.
+
+Instead, you could use `crane rebase` to replace the vulnerable base image
+layers in your image with new patched base image layers from the newly released
+base image, without needing to rebuild from source, or indeed have access to
+the source at all.
+
+```
+$ crane rebase \
+  --original=gcr.io/my-project/my-app:latest \
+  --old_base=ubuntu@sha256:deadbeef... \
+  --new_base=ubuntu:latest \
+  --rebased=gcr.io/my-project/my-app:rebased
+```
+
+This command would fetch the manifest for `original`, `old_base` and `new_base`,
+check that `old_base` is indeed the basis for `original`, remove `old_base`'s
+layers from `original` and replace them with `new_base`'s layers, then compute
+and upload a new valid manifest for the image, tagged as `rebased`.
+
+If the image is in Google Container Registry, you can determine `old_base` image
+digests using `gcloud alpha container images describe <image>
+--show-image-basis`.
+
+## Rebase visualized
+
+![rebase visualization](./rebase.png)
```

Comment:
Can you put a link to the diagram in a comment and share it with the usual suspects?

Reply:
(see above) I'd also move this up